.github/workflows/cross-arch-test.yaml

```
@@ -0,0 +1,37 @@
+---
+name: Cross-Platform Tests
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  cross-test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        goos: [linux]
+        goarch: [arm, mips, mipsle, mips64, mips64le, ppc64, ppc64le, s390x]
+
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      - id: goversion
+        run: echo "goversion=$(cat .go-version)" >> "$GITHUB_OUTPUT"
+      - uses: actions/setup-go@d35c59abb061a4a6fb18e82ac0862c26744d6ab5 # v5.5.0
+        with:
+          go-version: ${{ steps.goversion.outputs.goversion }}
+
+      - name: Setup binfmt
+        uses: docker/setup-qemu-action@29109295f81e9208d7d86ff1c6c12d2833863392 # v3.6.0
+        with:
+          platforms: all
+
+      - name: Build for ${{ matrix.goos }}/${{ matrix.goarch }}
+        run: |
+          GOOS=${{ env.GOOS }} GOARCH=${{ env.GOARCH }} go build ./...
```

It seems that you are using `matrix.goos` and `matrix.goarch` at all
```suggestion
          GOOS=${{ matrix.goos }} GOARCH=${{ matrix.goarch }} go build ./...
```

---

fixed 👍🏽